### PR TITLE
[DO_NOT_MERGE] Remove Login Using User Id And Add Login Using UserName

### PR DESCRIPTION
**Pull request by nanopay-arthur**
Ref: https://nanopay.atlassian.net/browse/CPF-4063

### CODE REVIEW

File: src/foam/nanos/auth/PasswordExpiryAuthService.js [hunk 1]
```diff
@@ -22,34 +22,39 @@ foam.CLASS({
...
-if ( isPasswordExpired(user) ) {
-  throw new AuthenticationException("Password expired");
-}
-return getDelegate().loginByEmail(x, email, password);`
+        `User user = (User) ((DAO) getLocalUserDAO())
+          .inX(x)
+          .find(
+            AND(
+              OR(
+                EQ(foam.nanos.auth.User.EMAIL, id.toLowerCase()),
```
Comment (TW80000) [inline src/foam/nanos/auth/PasswordExpiryAuthService.js]: Pretty sure we don't need to include the package for `User` here and on lines 48 and 50.

File: src/foam/nanos/auth/PasswordExpiryAuthService.js [hunk 2]
```diff
@@ -22,34 +22,39 @@ foam.CLASS({
...
+                EQ(foam.nanos.auth.User.USER_NAME, id)
+              ),
+              EQ(foam.nanos.auth.User.LOGIN_ENABLED, true)
```
Comment (TW80000) [inline src/foam/nanos/auth/PasswordExpiryAuthService.js]: Should we be checking any of these properties on User as well?

* enabled (deprecated)
* emailVerified
* status
* deleted (will be deprecated soon)
* lifecycleState (nanopay side)

File: src/foam/nanos/u2/navigation/SignIn.js [hunk 3]
```diff
@@ -98,27 +100,32 @@ foam.CLASS({
...
-        ).catch(
-          (err) => {
-            this.notify(err.message || 'There was a problem logging in.', 'error');
-        });
+          ).catch(
+            (err) => {
+              this.notify(err.message || 'There was a problem logging in.', 'error');
+          });
+        } else {
+          //TODO: change to 'Please enter email or username' when integratting
```
Comment (TW80000) [inline src/foam/nanos/u2/navigation/SignIn.js]:
```suggestion
          //TODO: change to 'Please enter email or username' when integrating
```

and several places elsewhere. Not very important though.

File: src/foam/nanos/auth/PasswordExpiryAuthService.js (hunk 2 above)
Comment (jlhughes) [reply src/foam/nanos/auth/PasswordExpiryAuthService.js]: emailVerified: AND ( EQ(email, id), EQ(emailverified, true)
status: this is compliance - don't know if we block login on compliance issues - no way for them to correct anything - please inquire with Nadir. 
lifecycle: EQ (active)

Comment (nanopay-arthur) [reply src/foam/nanos/auth/PasswordExpiryAuthService.js]: enabled: I will remove this.

emailVerified: We currently allow users with an unverified email to login to Ablii. They just get stuck at the "We sent you and email, please verify you email" page, so I don't think we should add the emailVerified condition. 

status: Checked in with Nadir and he said that we should prevent a user from logging in if their status is set to "Disabled" or "Revoked". Check already exists on NanopayUserAndGroupAuthService decorator.

lifecycle: don't add this. (doesn't exist on dev yet)

**Pull request closed without merging**